src/storage/src/hummock/sstable/block.rs

```diff
@@ -61,7 +61,7 @@ impl Block {
             CompressionAlgorithm::Zstd => {
                 let mut decoder = zstd::Decoder::new(compressed_data.reader())
                     .map_err(HummockError::decode_error)?;
-                let mut decoded = Vec::with_capacity(DEFAULT_BLOCK_SIZE);
+                let mut decoded = Vec::with_capacity(uncompressed_capacity);
```

Comment:
We expect the incoming uncompressed_capacity to be correct, in order to ensure this point, can we add a check ? Because the error parm can also be executed, we can found the problem ASAP

Reply:
done.